data/arithmetics_dataset/expression_generator.py

```diff
@@ -0,0 +1,122 @@
+import random
+
+
+class ExpressionGenerator:
+    """
+    Generates random arithmetic expressions with specified properties.
+    """
+
+    def __init__(
+        self,
+        operators: list[str] = None,
+        negative_probability: float = 0.5,
+        parentheses_probability: float = 1.0,
+    ):
+        """
+        Initializes the generator with operators, number digit limits, and probabilities.
+
+        Parameters:
+            operators: A list of operators to use in the expressions. Defaults to ["+", "-", "*"].
+            negative_probability: The probability that a number is negative.
+            parentheses_probability: The probability that parts of the expression are wrapped in parentheses.
+        """
+        self._operators = operators or ["+", "-", "*"]
```

Comment:
I would move the `["+", "-", "*"]` to the signature, otherwise strictly speaking the type annotation should be `Optional[List[str]]` but because you have an actual default it's better to move it to the signature explicitly

Reply:
I was worried about using a mutable type as a default value of a function parameter. As the value itself is never altered I guess it would be fine to use. I would prefer to change the type annotation to `Optional[List[str]]` though to avoid any potential issues.